[BUG] Cannot change icons when customizing app icon in work profile

### Describe the bug
Lawnchair either does not load the icon or crashes when changing app icons for applications in the Work profile.

### Steps to reproduce
1. Select and hold an icon that has a work profile badge
2. Click 'Customize' in the pop-up menu
3. Edit the app icon by clicking the logo of the app
4. Select any icon on System Icons or in your installed Icon Pack
5. App either crashes, or does nothing.

### Expected behavior

It should not crash and allow to change the icon just like it does in the Personal profile.

### Screenshots

Based on #3699:

[Video](https://github.com/LawnchairLauncher/lawnchair/assets/14168726/0e158bcd-a92c-4064-9574-4b9b03a89945)

### Device information

Pixel 8 Pro, GrapheneOS

### Additional context

```
type: crash
osVersion: google/husky/husky:15/AP3A.241105.007/2024110700:user/release-keys
package: app.lawnchair:14000203, targetSdk 34
process: app.lawnchair
processUptime: 17745 + 184 ms
installer: dev.imranr.obtainium

java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.CharSequence android.content.pm.LauncherActivityInfo.getLabel()' on a null object reference
	at m8.h6.d(Unknown Source:78)
	at n8.a$p.a(Unknown Source:58)
	at n8.a$p.c(Unknown Source:12)
	at j1.b.d(Unknown Source:44)
	at j1.b.c(Unknown Source:8)
	at p5.m$k$a.a(Unknown Source:57)
	at p5.m$k$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.a(Unknown Source:30)
	at l1.e.d(Unknown Source:174)
	at p5.j.b(Unknown Source:139)
	at p5.j.c(Unknown Source:0)
	at p5.j$a.a(Unknown Source:36)
	at p5.j$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.b(Unknown Source:30)
	at p5.j.a(Unknown Source:129)
	at p5.m$k.a(Unknown Source:84)
	at p5.m$k.c(Unknown Source:12)
	at j1.b.d(Unknown Source:44)
	at j1.b.c(Unknown Source:8)
	at androidx.compose.animation.a$c$d.a(Unknown Source:165)
	at androidx.compose.animation.a$c$d.invoke(Unknown Source:10)
	at j1.b.b(Unknown Source:44)
	at j1.b.invoke(SourceFile:2)
	at y.e.a(Unknown Source:876)
	at androidx.compose.animation.a$c.a(Unknown Source:317)
	at androidx.compose.animation.a$c.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at androidx.compose.animation.a.a(Unknown Source:887)
	at p5.m.b(Unknown Source:1211)
	at p5.m.a(Unknown Source:612)
	at n8.n.l(Unknown Source:338)
	at c8.n$d$a$a$a.a(Unknown Source:27)
	at c8.n$d$a$a$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at c8.n.j(Unknown Source:173)
	at c8.n.p(Unknown Source:0)
	at c8.n$d$a$a.a(Unknown Source:46)
	at c8.n$d$a$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.b(Unknown Source:30)
	at c8.n$d$a.a(Unknown Source:86)
	at c8.n$d$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at x0.f2$b.a(Unknown Source:212)
	at x0.f2$b.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.b(Unknown Source:30)
	at x0.f2.a(Unknown Source:177)
	at c8.n$d.a(Unknown Source:90)
	at c8.n$d.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at c8.n$e$a.a(Unknown Source:23)
	at c8.n$e$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at q8.w$a.b(Unknown Source:28)
	at q8.w$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.a(Unknown Source:30)
	at q8.w.j(Unknown Source:278)
	at c8.n$e.a(Unknown Source:34)
	at c8.n$e.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at r8.o0$a.a(Unknown Source:23)
	at r8.o0$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.a(Unknown Source:30)
	at r8.o0.f(Unknown Source:74)
	at c8.n.n(Unknown Source:55)
	at c8.n.l(Unknown Source:395)
	at app.lawnchair.ui.preferences.PreferenceActivity$b$a.a(Unknown Source:29)
	at app.lawnchair.ui.preferences.PreferenceActivity$b$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.a(Unknown Source:30)
	at x0.q2.a(Unknown Source:89)
	at x0.p0$b.a(Unknown Source:40)
	at x0.p0$b.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.b(Unknown Source:30)
	at x0.p0.a(Unknown Source:289)
	at p8.g.f(Unknown Source:132)
	at app.lawnchair.ui.preferences.PreferenceActivity$b.a(Unknown Source:42)
	at app.lawnchair.ui.preferences.PreferenceActivity$b.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at o2.q0.a(Unknown Source:82)
	at o2.a$a.a(Unknown Source:34)
	at o2.a$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.b(Unknown Source:30)
	at o2.r0.a(Unknown Source:310)
	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt$i.a(Unknown Source:38)
	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt$i.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.b(Unknown Source:30)
	at androidx.compose.ui.platform.AndroidCompositionLocals_androidKt.a(Unknown Source:356)
	at androidx.compose.ui.platform.k$a$a$c.a(Unknown Source:40)
	at androidx.compose.ui.platform.k$a$a$c.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.x.a(Unknown Source:30)
	at androidx.compose.ui.platform.k$a$a.a(Unknown Source:221)
	at androidx.compose.ui.platform.k$a$a.invoke(Unknown Source:8)
	at j1.b.a(Unknown Source:44)
	at j1.b.invoke(SourceFile:1)
	at b1.c.d(Unknown Source:17)
	at b1.n.r0(Unknown Source:236)
	at b1.n.k0(Unknown Source:13)
	at b1.t.v(Unknown Source:15)
	at b1.n2.a(Unknown Source:24)
	at b1.t.B(Unknown Source:15)
	at b1.t.z(Unknown Source:0)
	at androidx.compose.ui.platform.k$a.a(Unknown Source:77)
	at androidx.compose.ui.platform.k$a.invoke(Unknown Source:2)
	at androidx.compose.ui.platform.g.J1(Unknown Source:6)
	at androidx.compose.ui.platform.k.z(Unknown Source:7)
	at androidx.compose.ui.platform.k.s(Unknown Source:18)
	at androidx.lifecycle.l$b.a(Unknown Source:24)
	at androidx.lifecycle.l.c(Unknown Source:105)
	at androidx.compose.ui.platform.k$a.a(Unknown Source:38)
	at androidx.compose.ui.platform.k$a.invoke(Unknown Source:2)
	at androidx.compose.ui.platform.g.onAttachedToWindow(Unknown Source:119)
	at android.view.View.dispatchAttachedToWindow(View.java:22868)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3492)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3499)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3499)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3499)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3499)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3499)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3499)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3443)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2917)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10324)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1406)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
	at android.view.Choreographer.doCallbacks(Choreographer.java:1015)
	at android.view.Choreographer.doFrame(Choreographer.java:945)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1389)
	at android.os.Handler.handleCallback(Handler.java:959)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8744)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
```

>     * I've tried reproducing the problem, and what I've noticed is that the "original" app's custom icon does change to whatever icon was chosen for the secondary app. For example, if I try adding a Clock icon to my **WORK** Play Store app, my **WORK** Play Store app's icon **remains the same**, but my **PERSONAL** Play Store app's icon **changes to the Clock icon**.

I don't even get far enough as to change the icon - I press Cuztomize then press the 'Edit' pencil on the icon to change it, and the crash occurs then before I get a chance to pick a new icon. The error is the second one you mentioned:
`java.lang.SecurityEsception: Cannot resolve activity for unrelated profile 0`

To reproduce:

- Install latest alpha version of GrapheneOS
- Create a second user profile and install Lawnchair 15.Beta 1 onto it
- From the second user profile, from within Lawnchair, long press any app icon on the home screen, select Customize
- Tap on the icon to edit it

Device: Pixel 9 Pro XL
Android version: 16
OS: GrapheneOS version 2025080600
Kernel version: 6.1.147-android14-11-gb99fadd977e1
Lawnchair version: 15.Beta 1

Full crash report:

```
Lawnchair bug report 7 Aug 2025 17:39:42
version: 15.Beta 1 (15000201)
commit: 4fd5960
build.brand: google
build.device: komodo
build.display: 2025080600
build.fingerprint: google/komodo/komodo:16/BP2A.250805.005/2025080600:user/release-keys
build.hardware: komodo
build.id: BP2A.250805.005
build.manufacturer: Google
build.model: Pixel 9 Pro XL
build.security.level: 2025-08-05
build.product: komodo
build.type: user
version.codename: REL
version.incremental: 2025080600
version.release: 16
version.sdk_int: 36
display.density_dpi: 480
isRecentsEnabled: false

error: Uncaught exception

java.lang.SecurityException: Cannot resolve activity for unrelated profile 0
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3349)
	at android.os.Parcel.createException(Parcel.java:3333)
	at android.os.Parcel.readException(Parcel.java:3309)
	at android.os.Parcel.readException(Parcel.java:3251)
	at android.content.pm.ILauncherApps$Stub$Proxy.resolveLauncherActivityInternal(ILauncherApps.java:1014)
	at android.content.pm.LauncherApps.resolveActivity(LauncherApps.java:986)
	at t9.w8.d(Unknown Source:126)
	at u9.m$s.a(Unknown Source:49)
	at u9.m$s.c(Unknown Source:12)
	at k1.c.d(Unknown Source:44)
	at k1.c.c(Unknown Source:8)
	at s5.o0$d$a.a(Unknown Source:57)
	at s5.o0$d$a.invoke(Unknown Source:8)
	at k1.c.a(Unknown Source:44)
	at k1.c.invoke(SourceFile:1)
	at c1.x.a(Unknown Source:30)
	at m1.e.f(Unknown Source:114)
	at s5.r.f(Unknown Source:166)
	at s5.r.i(Unknown Source:0)
	at s5.r$a.a(Unknown Source:36)
	at s5.r$a.invoke(Unknown Source:8)
	at k1.c.a(Unknown Source:44)
	at k1.c.invoke(SourceFile:1)
	at c1.x.b(Unknown Source:30)
	at s5.r.d(Unknown Source:129)
	at s5.o0$d.a(Unknown Source:113)
	at s5.o0$d.c(Unknown Source:12)
	at k1.c.d(Unknown Source:44)
	at k1.c.c(Unknown Source:8)
	at androidx.compose.animation.a$c$d.a(Unknown Source:164)
	at androidx.compose.animation.a$c$d.invoke(Unknown Source:10)
	at k1.c.b(Unknown Source:44)
	at k1.c.invoke(SourceFile:2)
	at y.e.a(Unknown Source:851)
	at androidx.compose.animation.a$c.a(Unknown Source:320)
	at androidx.compose.animation.a$c.invoke(Unknown Source:8)
	at k1.c.a(Unknown Source:44)
	at k1.c.invoke(SourceFile:1)
	at androidx.compose.animation.a.a(Unknown Source:844)
	at s5.o0.r(Unknown Source:1305)
	at s5.o0.q(Unknown Source:711)
	at u9.d1.g(Unknown Source:334)
	at i9.n$e$a$a$a.a(Unknown Source:36)
	at i9.n$e$a$a$a.invoke(Unknown Source:8)
	at k1.c.a(Unknown Source:44)
	at k1.c.invoke(SourceFile:1)
	at i9.n$d.a(Unknown Source:38)
	at i9.n$d.invoke(Unknown Source:8)
	at k1.c.a(Unknown Source:44)
	at k1.c.invoke(SourceFile:1)
	at c1.k1$b.a(Unknown Source:22)
	at c1.k1$b.invoke(Unknown Source:10)
	at k1.c.b(Unknown Source:44)
	at k1.c.invoke(SourceFile:2)
	at c1.m$f.a(Unknown Source:44)
	at c1.m$f.invoke(Unknown Source:8)
	at k1.c.a(Unknown Source:44)
	at k1.c.invoke(SourceFile:1)
	at k1.s.b(Unknown Source:17)
	at c1.m.O0(Unknown Source:144)
	at c1.m.e0(Unknown Source:0)
	at c1.m$d.a(Unknown Source:68)
	at c1.m$d.invoke(Unknown Source:0)
	at c1.m.X0(Unknown Source:63)
	at c1.m.Y0(Unknown Source:24)
	at c1.m.L0(Unknown Source:133)
	at c1.m.M0(Unknown Source:0)
	at c1.t.g(Unknown Source:44)
	at c1.p2.r0(Unknown Source:439)
	at c1.p2.p0(Unknown Source:46)
	at c1.p2.a(Unknown Source:78)
	at c1.t.x(Unknown Source:13)
	at c1.t.e(Unknown Source:0)
	at androidx.compose.ui.platform.j$a.a(Unknown Source:77)
	at androidx.compose.ui.platform.j$a.invoke(Unknown Source:2)
	at androidx.compose.ui.platform.AndroidComposeView.W1(Unknown Source:6)
	at androidx.compose.ui.platform.j.e(Unknown Source:7)
	at androidx.compose.ui.platform.j.z(Unknown Source:18)
	at androidx.lifecycle.k$b.a(Unknown Source:24)
	at androidx.lifecycle.k.c(Unknown Source:105)
	at androidx.compose.ui.platform.j$a.a(Unknown Source:38)
	at androidx.compose.ui.platform.j$a.invoke(Unknown Source:2)
	at androidx.compose.ui.platform.AndroidComposeView.onAttachedToWindow(Unknown Source:132)
	at android.view.View.dispatchAttachedToWindow(View.java:23105)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3520)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3617)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:3076)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10643)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1570)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1579)
	at android.view.Choreographer.doCallbacks(Choreographer.java:1179)
	at android.view.Choreographer.doFrame(Choreographer.java:1108)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1553)
	at android.os.Handler.handleCallback(Handler.java:995)
	at android.os.Handler.dispatchMessage(Handler.java:103)
	at android.os.Looper.loopOnce(Looper.java:248)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:9106)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:593)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:371)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.pm.UserManagerService$LocalService.isProfileAccessible(UserManagerService.java:8221)
	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.canAccessProfile(LauncherAppsService.java:520)
	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.canAccessProfile(LauncherAppsService.java:497)
	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.resolveLauncherActivityInternal(LauncherAppsService.java:816)
	at android.content.pm.ILauncherApps$Stub.onTransact(ILauncherApps.java:429)

```

I'm seeing same issue, but not in Work profile, but a separate/secondary profile (e.g. GrapheneOS user profiles, device Pixel 9a).

NOTE: I does work as expected on the Owner profile.

Tested on both 15-beta1 (https://github.com/LawnchairLauncher/lawnchair/commit/4fd596016bad5cf2504da14e8520dfa7cde57e01) and nightly (https://github.com/LawnchairLauncher/lawnchair/commit/67b9051a554b592395156540c5c0a40c6f3e892f).

```
type: crash
osVersion: google/tegu/tegu:16/BP2A.250805.005/2025102801:user/release-keys
userType: full.secondary
package: app.lawnchair.nightly:15000201, targetSdk 35
process: app.lawnchair.nightly
processUptime: 24692 + 300 ms

java.lang.SecurityException: Cannot resolve activity for unrelated profile 0
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3348)
	at android.os.Parcel.createException(Parcel.java:3332)
	at android.os.Parcel.readException(Parcel.java:3308)
	at android.os.Parcel.readException(Parcel.java:3250)
	at android.content.pm.ILauncherApps$Stub$Proxy.resolveLauncherActivityInternal(ILauncherApps.java:1015)
	at android.content.pm.LauncherApps.resolveActivity(LauncherApps.java:986)
	at da.uj.g(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:124)
	at ea.s0.v0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:50)
	at ea.s0.f(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at ea.w.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:13)
	at t1.h.l(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at c6.o0$d$a.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:58)
	at c6.o0$d$a.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at t1.h.o(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at l1.h0.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:31)
	at v1.m.c(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:198)
	at c6.r.f(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:167)
	at c6.r.i(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at c6.r$a.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:37)
	at c6.r$a.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at t1.h.o(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at l1.h0.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:31)
	at c6.r.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:130)
	at c6.o0$d.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:114)
	at c6.o0$d.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:13)
	at t1.h.l(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.d(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at androidx.compose.animation.a$c$d.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:165)
	at androidx.compose.animation.a$c$d.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:11)
	at t1.h.m(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:2)
	at y.e.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:856)
	at androidx.compose.animation.a$c.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:321)
	at androidx.compose.animation.a$c.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at t1.h.o(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at androidx.compose.animation.a.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:885)
	at c6.o0.q(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:121)
	at c6.o0.r(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:26)
	at ea.m2.i(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:289)
	at s9.y.L(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:33)
	at s9.y.e(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at s9.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:13)
	at t1.h.o(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at s9.y.x(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:37)
	at s9.y.g(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at s9.i.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:11)
	at t1.h.o(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at l1.w1$b.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:23)
	at l1.w1$b.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:11)
	at t1.h.m(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:2)
	at l1.s$d.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at l1.s$d.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at t1.h.o(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at t1.h.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at t1.j.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:18)
	at l1.s.V0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:146)
	at l1.s.R0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:46)
	at l1.s.c0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at l1.n.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:9)
	at l1.s.g1(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:64)
	at l1.s.h1(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:25)
	at l1.s.Q0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:134)
	at l1.s.T0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:1)
	at l1.b0.f(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:45)
	at l1.p3.C0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:440)
	at l1.p3.A0(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:47)
	at l1.p3.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:122)
	at l1.b0.F(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:5)
	at l1.b0.m(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:14)
	at androidx.compose.ui.platform.k$a.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:78)
	at androidx.compose.ui.platform.k$a.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:3)
	at androidx.compose.ui.platform.AndroidComposeView.f2(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:7)
	at androidx.compose.ui.platform.k.m(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:8)
	at androidx.compose.ui.platform.k.A(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:19)
	at androidx.lifecycle.k$b.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:25)
	at androidx.lifecycle.k.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:106)
	at androidx.compose.ui.platform.k$a.a(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:39)
	at androidx.compose.ui.platform.k$a.invoke(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:3)
	at androidx.compose.ui.platform.AndroidComposeView.onAttachedToWindow(r8-map-id-9f8e05c8efc50c99a59552cce9f48b2778dbaa6551103a0e68ad0ecc438b901c:162)
	at android.view.View.dispatchAttachedToWindow(View.java:23105)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3520)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3527)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3617)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:3076)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:10643)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1570)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1579)
	at android.view.Choreographer.doCallbacks(Choreographer.java:1179)
	at android.view.Choreographer.doFrame(Choreographer.java:1108)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1553)
	at android.os.Handler.handleCallback(Handler.java:995)
	at android.os.Handler.dispatchMessage(Handler.java:103)
	at android.os.Looper.loopOnce(Looper.java:248)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:9111)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:593)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:371)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.pm.UserManagerService$LocalService.isProfileAccessible(UserManagerService.java:8221)
	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.canAccessProfile(LauncherAppsService.java:520)
	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.canAccessProfile(LauncherAppsService.java:497)
	at com.android.server.pm.LauncherAppsService$LauncherAppsImpl.resolveLauncherActivityInternal(LauncherAppsService.java:816)
	at android.content.pm.ILauncherApps$Stub.onTransact(ILauncherApps.java:429)
```

--- a/lawnchair/src/app/lawnchair/icons/LawnchairIconProvider.kt
+++ b/lawnchair/src/app/lawnchair/icons/LawnchairIconProvider.kt
@@ -14,6 +14,8 @@ import android.content.Intent.ACTION_TIME_CHANGED
 import android.content.Intent.ACTION_TIME_TICK
 import android.content.IntentFilter
 import android.content.pm.ApplicationInfo
+import android.content.pm.ComponentInfo
+import android.content.pm.LauncherApps
 import android.content.pm.PackageItemInfo
 import android.content.res.Resources
 import android.graphics.drawable.Drawable
@@ -32,6 +34,7 @@ import app.lawnchair.icons.picker.IconType
 import app.lawnchair.preferences.PreferenceManager
 import app.lawnchair.util.MultiSafeCloseable
 import app.lawnchair.util.isPackageInstalled
+import app.lawnchair.util.requireSystemService
 import com.android.launcher3.R
 import com.android.launcher3.dagger.ApplicationContext
 import com.android.launcher3.dagger.LauncherAppSingleton
@@ -107,14 +110,42 @@ class LawnchairIconProvider @Inject constructor(
         return iconPack.getIcon(componentName)
     }
 
+    /**
+     * Resolves the launch component for icon-pack lookup.
+     *
+     * Avoid [android.content.pm.PackageManager.getLaunchIntentForPackage], which only sees the
+     * current user — work/private profile apps would otherwise fall back to the system icon.
+     * Prefer the real component from [ComponentInfo], then [LauncherApps] for the app's user.
+     */
+    private fun resolveComponentName(
+        info: PackageItemInfo,
+        appInfo: ApplicationInfo,
+        user: UserHandle,
+    ): ComponentName? {
+        if (info is ComponentInfo && !info.name.isNullOrEmpty()) {
+            return ComponentName(info.packageName ?: appInfo.packageName, info.name)
+        }
+        return resolveLaunchComponent(appInfo.packageName, user)
+    }
+
+    private fun resolveLaunchComponent(packageName: String, user: UserHandle): ComponentName? {
+        return try {
+            val launcherApps: LauncherApps = context.requireSystemService()
+            launcherApps.getActivityList(packageName, user).firstOrNull()?.componentName
+        } catch (t: Throwable) {
+            Log.w(TAG, "Failed to resolve launch component for $packageName user=$user", t)
+            null
+        }
+    }
+
     override fun getIcon(
         info: PackageItemInfo,
         appInfo: ApplicationInfo,
         iconDpi: Int,
     ): Drawable {
         val packageName = appInfo.packageName
-        val componentName = context.packageManager.getLaunchIntentForPackage(packageName)?.component
         val user = UserHandle.getUserHandleForUid(appInfo.uid)
+        val componentName = resolveComponentName(info, appInfo, user)
 
         var iconEntry: IconEntry? = null
         if (componentName != null) {

--- a/lawnchair/src/app/lawnchair/icons/iconpack/SystemIconPack.kt
+++ b/lawnchair/src/app/lawnchair/icons/iconpack/SystemIconPack.kt
@@ -26,7 +26,13 @@ class SystemIconPack(context: Context, pkg: String) : IconPack(context, pkg) {
         val profiles = UserCache.INSTANCE.get(context).userProfiles
         val launcherApps: LauncherApps = context.requireSystemService()
         profiles
-            .flatMap { launcherApps.getActivityList(null, Process.myUserHandle()) }
+            .flatMap { profile ->
+                try {
+                    launcherApps.getActivityList(null, profile)
+                } catch (_: Throwable) {
+                    emptyList()
+                }
+            }
             .associateBy { ComponentKey(it.componentName, it.user) }
     }
 
@@ -34,7 +40,11 @@ class SystemIconPack(context: Context, pkg: String) : IconPack(context, pkg) {
         startLoad()
     }
 
-    override fun getIcon(componentName: ComponentName) = IconEntry(packPackageName, ComponentKey(componentName, Process.myUserHandle()).toString(), IconType.Normal)
+    override fun getIcon(componentName: ComponentName): IconEntry {
+        val matching = appMap.keys.firstOrNull { it.componentName == componentName }
+        val key = matching ?: ComponentKey(componentName, Process.myUserHandle())
+        return IconEntry(packPackageName, key.toString(), IconType.Normal)
+    }
     override fun getCalendar(componentName: ComponentName): IconEntry? = null
     override fun getClock(entry: IconEntry): ClockMetadata? = null
 

--- a/lawnchair/src/app/lawnchair/ui/preferences/destinations/SelectIconPreference.kt
+++ b/lawnchair/src/app/lawnchair/ui/preferences/destinations/SelectIconPreference.kt
@@ -113,12 +113,13 @@ private fun resolveAppLabel(launcherApps: LauncherApps, componentKey: ComponentK
         Log.w(TAG, "resolveActivity failed for $componentKey", t)
     }
     try {
-        launcherApps.getActivityList(componentName.packageName, user)
+        val activities = launcherApps.getActivityList(componentName.packageName, user)
+        activities
             .firstOrNull { it.componentName == componentName }
             ?.label
             ?.toString()
             ?.let { return it }
-        launcherApps.getActivityList(componentName.packageName, user)
+        activities
             .firstOrNull()
             ?.label
             ?.toString()

--- a/lawnchair/src/app/lawnchair/ui/preferences/destinations/SelectIconPreference.kt
+++ b/lawnchair/src/app/lawnchair/ui/preferences/destinations/SelectIconPreference.kt
@@ -3,6 +3,7 @@ package app.lawnchair.ui.preferences.destinations
 import android.app.Activity
 import android.content.Intent
 import android.content.pm.LauncherApps
+import android.util.Log
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.remember
@@ -27,14 +28,13 @@ import com.android.launcher3.R
 import com.android.launcher3.util.ComponentKey
 import kotlinx.coroutines.launch
 
+private const val TAG = "SelectIconPreference"
+
 @Composable
 fun SelectIconPreference(componentKey: ComponentKey) {
     val context = LocalContext.current
     val label = remember(componentKey) {
-        val launcherApps: LauncherApps = context.requireSystemService()
-        val intent = Intent().setComponent(componentKey.componentName)
-        val activity = launcherApps.resolveActivity(intent, componentKey.user)
-        activity.label.toString()
+        resolveAppLabel(context.requireSystemService(), componentKey)
     }
     val iconPacks by LocalPreferenceInteractor.current.iconPacks.collectAsStateWithLifecycle()
     val navController = LocalNavController.current
@@ -95,4 +95,36 @@ fun SelectIconPreference(componentKey: ComponentKey) {
             )
         }
     }
+}
+
+/**
+ * Resolves a display label for [componentKey], including work/private profiles.
+ *
+ * [LauncherApps.resolveActivity] can return null or throw for non-current users; fall back to
+ * [LauncherApps.getActivityList] and finally the activity class name.
+ */
+private fun resolveAppLabel(launcherApps: LauncherApps, componentKey: ComponentKey): String {
+    val componentName = componentKey.componentName
+    val user = componentKey.user
+    try {
+        val intent = Intent().setComponent(componentName)
+        launcherApps.resolveActivity(intent, user)?.label?.toString()?.let { return it }
+    } catch (t: Throwable) {
+        Log.w(TAG, "resolveActivity failed for $componentKey", t)
+    }
+    try {
+        launcherApps.getActivityList(componentName.packageName, user)
+            .firstOrNull { it.componentName == componentName }
+            ?.label
+            ?.toString()
+            ?.let { return it }
+        launcherApps.getActivityList(componentName.packageName, user)
+            .firstOrNull()
+            ?.label
+            ?.toString()
+            ?.let { return it }
+    } catch (t: Throwable) {
+        Log.w(TAG, "getActivityList failed for $componentKey", t)
+    }
+    return componentName.shortClassName?.trimStart('.') ?: componentName.packageName
 }

--- a/src/com/android/launcher3/model/data/ItemInfoWithIcon.java
+++ b/src/com/android/launcher3/model/data/ItemInfoWithIcon.java
@@ -16,7 +16,6 @@
 
 package com.android.launcher3.model.data;
 
-import static com.android.launcher3.icons.BitmapInfo.FLAG_NO_BADGE;
 import static com.android.launcher3.icons.BitmapInfo.FLAG_THEMED;
 
 import android.content.Context;
@@ -324,7 +323,8 @@ public abstract class ItemInfoWithIcon extends ItemInfo {
      */
     public FastBitmapDrawable newIcon(Context context) {
         var shouldTheme = PreferenceManager.getInstance(context).getThemedIcons().get();
-        return newIcon(context, shouldTheme ? BitmapInfo.FLAG_THEMED : BitmapInfo.FLAG_NO_BADGE);
+        // Use 0 (not FLAG_NO_BADGE) when theming is off so work/clone profile badges still draw.
+        return newIcon(context, shouldTheme ? FLAG_THEMED : 0);
     }
 
     /**

